tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,352 @@
+    if (script_run("[[ -f $standalone_path ]]") == 0){assert_script_run("source $standalone_path",60);}
+
+}
+
+sub destroy_standalone {
+
+    my $cleanup_path  = "/usr/share/qa/qa_test_virtualization/cleanup";
+    if (script_run("[[ -f $cleanup_path ]]") == 0){assert_script_run("source $cleanup_path",60);}
+
+}
+
+sub restart_libvirtd {
+
+    if (is_sle('>11')) {systemctl 'restart libvirtd';} else {script_run("service libvirtd restart");}
+
+}
+
+sub restore_guests {
+
+    my $wait_script        = "30";
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define $_.xml");
+	upload_logs "$_.xml";
+        script_run("rm -rf $_.xml");
+    }
+
+}
+
+sub restore_network {
+
+    my $network_mark  = "/etc/sysconfig/network/ifcfg-br0.new";
+    if (script_run("[[ -f $network_mark ]]") == 0){
+        assert_script_run("ip link set dev $virt_host_bridge down",60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+systemctl restart network.service;
+EOF
+         script_output($bridge_destroy, 900);
+    }
+
+}
+
+sub destroy_vir_network {
+
+    my $wait_script       = "30";
+    my $get_vnet_name     = "virsh net-list --all| tail -2| head -1| awk \'{print \$1}\'";
+    my $vnet_name         = script_output($get_vnet_name, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vnet_name_array = split(/\n+/, $vnet_name);
+    foreach (@vnet_name_array) {script_run("virsh net-destroy $_"); script_run("rm -rf $_.xml");}
+
+    my $default_path  = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0){assert_script_run("virsh net-define $default_path",60);assert_script_run("rm -rf $default_path");}
+
+}
+
+sub run {
+
+    #Prepare VM HOST SERVER Network Interface Configuration
+    #for libvirt virtual network testing 
+    prepare_network;
+
+    #Debug br123
+    restore_standalone;
+
+    #Remove SSH old files and Generate the key pair
+    assert_script_run "rm -rf ~/.ssh/* || true;ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa";
+ 
+    #Install required packages
+    zypper_call '-t in iptables iputils bind-utils sshpass';
+
+    #Guest SSH Setup
+    my $gi_guest = '';
+    foreach my $guest (keys %xen::guests) {
+        #Archive deployed Guests
+        assert_script_run("virsh dumpxml $guest > $guest.xml");
+        record_info "$guest", "Establishing SSH connection to $guest";
```

Comment:
Deployed Guest? 
after called tests/virt_autotest/guest_installation_run.pm, there will be established required Guests for virtual network testing.
As the verification run, there was deployed Guests on VM server for tests/virt_autotest/virsh_virtual_network.pm

Reply:
Here you are getting the xml, then you starting the guest and then you fetching the IP of the guest from the journal, is that correct? If so, my point is that your `record_info` is not valid because you are not establishing any SSH connection over here.

Reply:
Got it. Will delete this "record_info" as your mentioned.

Reply:
And please don't delete your comments 😄 Now it looks like me explaining something to myself...

Reply:
Opps, got it. 
I just find that my post feedback does not answer your comments. smile